### DISCUSSION TITLE
How to get around mutate onSuccess only being called for the last mutation?

### DISCUSSION THREAD

**Discussion by dmarshall83** [TanStack/query#2614] [Q&A] [answered]
I understand per the docs that for useMutation.mutate:

> If you make multiple requests, onSuccess will fire only after the latest call you've made.

I have a custom hook were I have used useQueries to get my data and then manage updates to the queries. One of these is save all which runs the mutation for anything that is out of sync. I use the onSuccess callback to update the sync information and it is only getting called for the last one. Is there any way for me to calls these differently or track the mutations individually in order to fire off the callbacks myself?  Do I need to dive into the mutation cache and watch for those updates somehow?

Any help would be greatly appreciated, thanks.

Dan

**Comment by TkDodo** [comment, marked as answer]
Hm, could be that this only applies to the callback on the `.mutate` function, _not_ the one for the `useMutation` call itself. Otherwise, maybe use `mutateAsync` to await the promise

**Reply by dmarshall83** [reply]
Thanks, I could have sworn that I tried mutateAsync but I must have done something wrong. I reimplemented with mutateAsync and put my callback in the "then" method on the promise and it worked fine. Thanks for the help.
